packages/react/src/components/button/Button.tsx

```diff
@@ -90,24 +108,31 @@ export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
       </div>
     ) : null;
 
+    const loadingOnClick: React.MouseEventHandler<HTMLButtonElement> = (event) => {
+      event.preventDefault();
+    };
+
     return (
       <button
         ref={ref}
         disabled={disabled}
+        aria-disabled={isLoading || disabled || undefined}
         type="button"
         className={classNames(
           styles.button,
           styles[variant],
           styles[`theme-${theme}`],
           styles[`size-${size}`],
           fullWidth ? styles.fullWidth : '',
+          isLoading ? styles.isLoading : '',
           className,
         )}
+        onClick={isLoading ? loadingOnClick : onClick}
```

Comment:
Was there no onClick before? onClick seems to be missing from the CommonButtonProps 🤔

Reply:
It is part of the react's button-props which ButtonProps extended.